test/topology_custom/test_change_rpc_ip.py

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""
+Test clusters can restart fine after an RPC IP address change.
+"""
+
+import time
+import logging
+
+import pytest
+
+from test.topology.util import wait_for_cql_and_get_hosts
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+async def two_node_cluster(manager):
+    logger.info(f"Booting initial 2-node cluster")
+    servers = [await manager.server_add(), await manager.server_add()]
+    for node in servers:
+        await manager.server_start(node.server_id)
+    await wait_for_cql_and_get_hosts(manager.get_cql(), servers, time.time() + 30)
+    return [node.server_id for node in servers]
+
+
+@pytest.mark.asyncio
+async def test_change_rpc_ip(two_node_cluster, manager, random_tables):
+    """Sequentially stop two nodes, change their RPC IPs and start, check the cluster is functional."""
+
+    table = await random_tables.add_table(ncolumns=5)
+
+    for node in reversed(two_node_cluster):  # change RPC IP address of the second node first.
```

Comment:
I could imagine more complex scenarios, for example, stopping both nodes and starting them with different ips, or changing both addresses at the same time (rpc and listen), or doing it along with changing dc/rack assignments.
Thoughts?